contracts/StakingVe.sol

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+/// @title A vesting style staking contract with extendable linear decay
+/// @author Auroter
+/// @notice Allows you to lock tokens in exchange for governance tokens
+/// @notice Locks can be extended or deposited into
+/// @notice Maximum deposit duration is two years (104 weeks)
+/// @dev If IsUserLocked is false, call CreateLock
+/// @dev If IsUserLocked is true, call ExtendMyLock
+
+// We have to maintain old versions of this to use legacy solidity (0.6.12)
+// Otherwise it's safe to link with @OpenZeppelin/
+import "./libraries/SafeERC20.sol";
+import "./libraries/SafeMath.sol";
+import "./interfaces/IERC20.sol";
+
+//import "hardhat/console.sol";
+
+contract StakingVe {
+    using SafeMath for uint256;
```

Comment:
Overflow/undeflow checks are implemented in solidity v0.8+, if we switch to that version we no longer need to use safe math

Reply:
yep